From 36026dd1197a21ea56fbbb43e34e246649dba4a7 Mon Sep 17 00:00:00 2001
From: kknitin <nitin@elk.audio>
Date: Wed, 12 Jun 2024 16:59:12 +0200
Subject: [PATCH] Fix for scarthgap gcc version

---
 CMakeLists.txt    | 2 +-
 src/raspa_pimpl.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ded3892..9acf615 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,7 +74,7 @@ else()
     add_xenomai_to_target(raspa)
 endif()
 
-target_include_directories(raspa PUBLIC ${PROJECT_SOURCE_DIR}/include)
+target_include_directories(raspa PUBLIC ${PROJECT_SOURCE_DIR}/include ${CMAKE_SYSROOT}/usr/include/uapi)
 target_include_directories(raspa PRIVATE ${PROJECT_SOURCE_DIR}/src/)
 set_property(TARGET raspa PROPERTY CXX_STANDARD 17)
 target_compile_options(raspa PRIVATE -Wall -Wextra -ffast-math -feliminate-unused-debug-types -fno-exceptions)
diff --git a/src/raspa_pimpl.h b/src/raspa_pimpl.h
index 9db186f..e915d09 100644
--- a/src/raspa_pimpl.h
+++ b/src/raspa_pimpl.h
@@ -57,6 +57,7 @@
 #include <memory>
 #include <string>
 #include <algorithm>
+#include <vector>
 
 #include "audio_control_protocol/audio_control_protocol.h"
 #include "audio_control_protocol/audio_packet_helper.h"
-- 
2.25.1

