From 7f6d90470d4528b3d660a45a4a402a8d9ea7829a Mon Sep 17 00:00:00 2001
From: kknitin <nitin@elk.audio>
Date: Wed, 15 Sep 2021 13:11:11 +0000
Subject: [PATCH] Remove setup_scm dependency

---
 setup.cfg | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index f2e9b29..4535e01 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -20,7 +20,6 @@ include_package_data = true
 python_requires = >=3.6
 install_requires =
 	netifaces
-setup_requires = setuptools_scm[toml] >= 3.4.1
 
 [options.packages.find]
 exclude =
-- 
2.17.1

